virtool/indexes/data.py

```diff
@@ -0,0 +1,303 @@
+            logger.debug("Index file upload aborted: %s", upload_id)
+            await delete_row(self._pg, upload_id, IndexFile)
+            return None
+
+        index_file = await virtool.uploads.db.finalize(
+            self._pg, size, upload_id, IndexFile
+        )
+
+        return IndexFile(**index_file)
+
+    async def finalize(self, ref_id: str, index_id: str) -> Index:
+        """
+        Finalize an index document.
+
+        :param ref_id: the reference ID
+        :param index_id: the index ID
+        :return: the finalized Index
+        """
+        reference = await self._db.references.find_one(ref_id)
+
+        if reference is None:
+            raise ResourceNotFoundError()
+
+        rows = await get_rows(self._pg, IndexFile, "index", index_id)
+
+        results = {f.name: f.type for f in rows}
+
+        if IndexType.fasta not in results.values():
+            raise ResourceConflictError(
+                "A FASTA file must be uploaded in order to finalize index"
+            )
+
+        if reference["data_type"] == "genome":
+            required_files = [f for f in FILES if f != "reference.json.gz"]
+
+            if missing_files := [f for f in required_files if f not in results]:
+                raise ResourceConflictError(
+                    f"Reference requires that all Bowtie2 index files have been uploaded. "
+                    f"Missing files: {', '.join(missing_files)}"
+                )
+
+        await virtool.indexes.db.finalize(
+            self._db, self._pg, self._config.base_url, ref_id, index_id
+        )
+
+        return await self.get(index_id)
+
+    async def find_changes(
+        self, index_id: str, req_query: MultiDictProxy[str]
+    ) -> HistorySearchResult:
+        """
+        Find the virus changes that are included in a given index build.
+
+        :param index_id: the index ID
+        :param req_query: the request query object
+        :return: the changes
+        """
+        if not await self._db.indexes.count_documents({"_id": index_id}):
+            raise ResourceNotFoundError()
+
+        db_query = {"index.id": index_id}
+
+        term = req_query.get("term")
+
+        if term:
+            db_query.update(compose_regex_query(term, ["otu.name", "user.id"]))
+
+        data = await paginate(
+            self._db.history,
+            db_query,
+            req_query,
+            sort=[("otu.name", 1), ("otu.version", -1)],
+            projection=LIST_PROJECTION,
+            reverse=True,
+        )
+
+        return HistorySearchResult(**data)
+
+    async def delete(self, index_id: str):
+        """
+        Delete an index given it's id.
+
+        :param index_id: the ID of the index to delete
+        """
+
+        delete_result = await self._db.indexes.delete_one({"_id": index_id})
+
+        if delete_result.deleted_count != 1:
+            # Document could not be deleted.
+            raise ResourceNotFoundError()
```

Comment:
Comment is redundant. Use simpler conditional checks whenever possible.
```suggestion
        if delete_result.deleted_count == 0:
            raise ResourceNotFoundError()
```